urldee901fae6_DocX_personal_timetabling_tgz-pJ8-net_personaltt_main_BenchmarkJ8/src/net/personaltt/main/Benchmark.java

```diff
@@ -0,0 +1,789 @@
+        //benchmarkProp(problem, "mainAllocationSelection.stayMinConflictSelectionAfterToZeroConflict", 5, 25, 5);
+        
+        //// benchmark probBestIgnoringSelWhenNoConflict
+        //benchmarkProp(problem, "mainAllocationSelection.probBestIgnoringSelWhenNoConflict", 0, 1, 0.2);
+        
+        //// benchmark optimizationWhenConflictProb
+        //benchmarkProp(problem, "rouletteOccurrenceSelection.optimizationWhenConflictProb", 0, 1, 0.2);
+       
+        //// benchmark priority
+        //benchmarkPriority();
+        
+        //benchmarkTerminationCondition();
+        
+        testAll();
+    }
+
+    public static void testAll() {
+        test1(100);
+        test2(100);
+        test3(100);
+        test4(100);
+        testS(1);
+        testP(50);
+        
+        test1(500);
+        test1(1000);
+        test1(5000);     
+        
+        test2(500);
+        test2(1000);
+        test2(5000);
+        
+        
+        test3(500);
+        test3(1000);
+        test3(5000);
+        
+        
+        test4(500);
+        test4(1000);
+        test4(5000);
+        
+        
+        testS(10);
+        testS(100);
+        testS(500);
+        
+        
+        testP(100);
+    }
+    
+    public static void test1(int number) {
+        System.out.printf("PROBLEM 1 - %s events\n", number);
+        ProblemDefinition p = bench1(number);
+        printInitialCosts(p);
+        testProblem(p);
+    }
+    
+    public static void test2(int number) {
+        System.out.printf("PROBLEM 2 - %s events\n", number);
+        ProblemDefinition p = bench2(number);
+        printInitialCosts(p);
+        testProblem(p);
+    }
+    
+    public static void test3(int number) {
+        System.out.printf("PROBLEM 3 - %s events\n", number);
+        ProblemDefinition p = bench3(number);
+        printInitialCosts(p);
+        testProblem(p);
+    }
+    
+    public static void test4(int number) {
+        System.out.printf("PROBLEM 4 - %s events\n", number);
+        ProblemDefinition p = bench4(number);
+        printInitialCosts(p);
+        testProblem(p);
+    }
+    
+    public static void testS(int number) {
+        System.out.printf("PROBLEM S - %s number\n", number);
+        ProblemDefinition p = bench5(number);
+        printInitialCosts(p);
+        testProblem(p);
+    }
+    
+    public static void testP(int number) {
+        System.out.printf("PROBLEM P - %s number\n", number);
+        ProblemDefinition p = benchPriority(1,number);
+        printInitialCosts(p);
+        testProblem(p);
+    }    
+
+    private static void printInitialCosts(ProblemDefinition p) {
+           
+        MainSolverState currentSolution = new MainSolverState();
+        currentSolution.init(p.initialSchedule);
+        
+        System.out.println(String.format("INITIAL COST: c:%s p:%s", currentSolution.constraintsCost(), currentSolution.preferenceCost()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: &#39;preferenceCost&#39; expression, &#39;constraintsCost&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 170. Consider formatting this data with &#39;%d&#39; instead.